kokkoro/modules/pcrclanbattle/clanbattle/cmdv2.py

```diff
@@ -904,22 +903,18 @@ async def boss_slayer(bot, ev: EventInterface, args: ParseResult):
         ext0 = 110 # 日服补偿刀20秒起
 
     remain = ev.get_param().remain
-    prm = re.findall("\d+[wW万]", remain)
-    if len(prm) == 2:
-        dmg1 = int(prm[0][:-1]) * 10000
-        dmg2 = int(prm[1][:-1]) * 10000
-    else:
-        prm = re.findall("\d+", remain)
-        if len(prm) == 2:
-            dmg1 = int(prm[0])
-            dmg2 = int(prm[1])
-        else:
-            usage = "【用法/用例】\n!补偿刀计算 50w 60w"
-            await bot.kkr_send(ev, usage, at_sender=True)
-            return
+    prm = re.findall("(\d+)([kK千]?)([wW万]?)", remain)
 
-    r, b, hp = bm.get_challenge_progress(1, datetime.now())
+    if len(prm) != 2:
+        usage = "【用法/用例】\n!补偿刀计算 伤害1 伤害2"
+        await bot.kkr_send(ev, usage, at_sender=True)
+        return
 
+    r, b, hp = bm.get_challenge_progress(datetime.now())
+    dmg1 = int(prm[0][0])
+    dmg2 = int(prm[1][0])
+    dmg1 = dmg1 * (1000 if prm[0][1] else 1) * (10000 if prm[0][2] else 1)
+    dmg2 = dmg2 * (1000 if prm[1][1] else 1) * (10000 if prm[1][2] else 1)
```

Comment:
Now we can deal with the damage number written in different formats at the same time